engine/flink/model-util/src/main/scala/pl/touk/nussknacker/engine/flink/util/transformer/aggregate/UnwrappingAggregateFunction.scala

```diff
@@ -1,54 +1,40 @@
 package pl.touk.nussknacker.engine.flink.util.transformer.aggregate
 
 import org.apache.flink.api.common.functions.AggregateFunction
-import org.apache.flink.api.java.tuple
-import org.apache.flink.api.java.tuple.Tuple2
 import pl.touk.nussknacker.engine.api.typed.typing.TypingResult
 import pl.touk.nussknacker.engine.api.{Context, ValueWithContext}
-import pl.touk.nussknacker.engine.flink.util.transformer.aggregate.UnwrappingAggregateFunction.AccumulatorWithContext
+import pl.touk.nussknacker.engine.flink.util.keyed.{KeyEnricher, StringKeyedValue}
 
 /**
- * This class unwraps value from input's KeyedValue. It also accumulate first Nussknacker's context that will be passed in output at the end.
+ * This class unwraps value from input's KeyedValue. It also accumulate key that will be passed in output at the end.
  *
  * NOTE: it would be much cleaner if we evaluated aggregateBy here. However, FLINK-10250 prevents us from doing this and we *have* to compute it beforehand
  *
  * When using this class it's important that agggregator, passedType and unwrap must match: unwrap result is of passedType and can be processed by aggregator
  */
-object UnwrappingAggregateFunction {
-  //We use Tuple2 here, to create TypeInformation more easily
-  type AccumulatorWithContext = Tuple2[AnyRef, Context]
-}
-
-class UnwrappingAggregateFunction[T](aggregator: Aggregator,
-                                     passedType: TypingResult,
-                                     unwrap: T => AnyRef,
-                                     outputContextStrategy: OutputContextStrategy)
-  extends AggregateFunction[ValueWithContext[T], Tuple2[AnyRef, Context], ValueWithContext[AnyRef]] {
+class UnwrappingAggregateFunction[Input](aggregator: Aggregator,
```

Comment:
I think we should use sth like: https://ci.apache.org/projects/flink/flink-docs-release-1.13/docs/dev/datastream/operators/windows/#incremental-window-aggregation-with-aggregatefunction to:
- avoid unnecessary key storing
- be able to add additional info (e.g. window start/stop) in the future